pkg/metadata/main.go

```diff
@@ -101,6 +116,14 @@ func main() {
 			netProviders = append(netProviders, NewVultr())
 		case p == "digitalocean":
 			netProviders = append(netProviders, NewDigitalOcean())
+		case p == "azure-imds":
+			netProviders = append(netProviders, NewAzureIMDS())
+		case p == "azure-ovf":
+			// TODO not every provider should create a separate http client
+			client := &http.Client{
+				Timeout: time.Second * 2,
+			}
+			netProviders = append(netProviders, NewAzureOVF(client))
```

Comment:
At the moment, each provider creates a separate http client, but [the docs state:](https://golang.org/pkg/net/http/)

> Clients and Transports are safe for concurrent use by multiple goroutines and for efficiency should only be created once and re-used.

Should I open a separate PR for this?

Reply:
I think we should entirely remove `azure-ovf` here, because `azure-imds` already uses it as a fallback.